core/procedures.ts

```diff
@@ -180,14 +180,21 @@ export function isNameUsed(
  * @alias Blockly.Procedures.rename
  */
 export function rename(this: Field, name: string): string {
+  const block = this.getSourceBlock();
+  if (!block) {
+    throw new Error(
+        'The field has not yet been attached to its input. ' +
```

Comment:
We're not doing this anywhere yet so maybe you don't want to do it willy-nilly.

but we could have a new error like UnattachedFieldError that is defined somewhere, with this message as default. Then you wouldn't have to keep repeating this string everywhere, and you can just `throw new UnattachedFieldError()` and even pass in the name of the field to the error message if you wanted.